delivery_postlogistics/postlogistics/web_service.py

```diff
@@ -78,7 +78,12 @@ def _prepare_recipient(self, picking):
         :return a dict containing data for ns0:Recipient
 
         """
-        partner = picking.partner_id
+        if picking.picking_type_id.code == "outgoing":
+            partner = picking.partner_id
+        elif picking.picking_type_id.code == "incoming":
+            location_dest = picking.location_dest_id
+            partner = location_dest.company_id.partner_id or self.env.user.company_id
```

Comment:
```suggestion
            partner = location_dest.company_id.partner_id or self.env.user.company_id.partner_id
```

But here the `partner` is not defined if the transfer is not `outgoing` or `incoming` (probably not real use cases but better to not crash if we want to use SwissPost on internal tranfers). So I would keep the original line as a default value and complete the code:
```python
partner = picking.partner_id
if picking.picking_type_id.code != "outgoing":
    location_dest = picking.location_dest_id
    partner = location_dest.company_id.partner_id or self.env.user.company_id.partner_id
```